### DISCUSSION TITLE
Resetting terminal cursor shape after exiting helix seems broken in 24.03

### DISCUSSION THREAD

**Discussion by wyw** [helix-editor/helix#10071] [Troubleshooting]
https://github.com/helix-editor/helix/assets/11427457/011b634b-89bd-4caf-b3bf-7f8b14980406

### Related issue & PR:
- https://github.com/helix-editor/helix/issues/2684
- https://github.com/helix-editor/helix/pull/8591
- [commits/553ffbc](https://github.com/helix-editor/helix/commits/553ffbcaa09847ec606539bddab8af1456450d73/helix-tui/src/backend/crossterm.rs)

Resetting the terminal cursor shape after exiting helix worked before the commit 553ffbc in version a2767269:

https://github.com/helix-editor/helix/assets/11427457/1d25ddfc-f02e-4c95-86e6-f8f2e4b2a50e

**Comment by archseer** [comment]
Converted to an issue, since this is a bug report and not a discussion. https://github.com/helix-editor/helix/issues/10089
